datafusion/spark/src/function/datetime/make_interval.rs

```diff
@@ -43,8 +43,39 @@ impl Default for SparkMakeInterval {
 
 impl SparkMakeInterval {
     pub fn new() -> Self {
+        fn int32_coercion() -> Coercion {
+            Coercion::new_implicit(
+                TypeSignatureClass::Native(logical_int32()),
+                vec![TypeSignatureClass::Integer],
+                NativeType::Int32,
+            )
+        }
+
+        fn float64_coercion() -> Coercion {
+            Coercion::new_implicit(
+                TypeSignatureClass::Native(logical_float64()),
+                vec![TypeSignatureClass::Numeric],
+                NativeType::Float64,
+            )
+        }
+
+        let mut variants = Vec::with_capacity(8);
+        variants.push(TypeSignature::Nullary);
+
+        for len in 1..=7 {
```

Comment:
I think I would prefer inlining the signature instead of generating it via this for loop logic; for example:

```rust
Signature::one_of(
    vec![
        TypeSignature::Nullary,
        // year
        TypeSignature::Coercible(vec![float64.clone()]),
        // year, month
        TypeSignature::Coercible(vec![float64.clone(), float64.clone()]),
        // etc.
    ],
    Volatility::Immutable,
)
```